lib/k8s/client/mint/request.ex

```diff
@@ -155,4 +174,46 @@ defmodule K8s.Client.Mint.Request do
       end
     end)
   end
+
+  @spec stream_request_body(t(), Mint.HTTP.t()) ::
+          {:ok, t(), Mint.HTTP.t()} | {:error, Mint.HTTP.t(), Mint.Types.error()}
+  def stream_request_body(%__MODULE__{mode: mode} = req, conn) when mode != :pending,
+    do: {:ok, req, conn}
+
+  def stream_request_body(request, conn) do
+    %__MODULE__{request_ref: request_ref, pending_request_body: pending_request_body} = request
+
+    {chunk, remaining_request_body} =
+      case Mint.HTTP.protocol(conn) do
+        :http1 ->
+          {pending_request_body, []}
+
+        :http2 ->
+          chunk_size = chunk_size(request, conn)
+          Enum.split(pending_request_body, chunk_size)
+      end
+
+    with {:ok, conn} <- Mint.HTTP.stream_request_body(conn, request_ref, chunk),
```

Comment:
Chunk here is not always the expected types of iodata. https://github.com/elixir-mint/mint/blob/v1.4.2/lib/mint/http2.ex#L508

When starting a watch on just about every kind on a cluster I'm seeing:
```
  * 1st argument: not an iodata term                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                               
            :erlang.iolist_size([123, 34, 97, 112, 105, 86, 101, 114, 115, 105, 111, 110, 34, 58, 34, 97, 112, 105, 101, 120, 116, 101, 110, 115, 105, 111, 110, 115, 46, 107, 56, 115, 46, 105, 111, 47, 118, 49, 34, 44, 34, 107, 105, 110, 100, 34, 58, 34, 67, 117, ...])                                  
            (mint 1.4.2) lib/mint/http2.ex:1196: Mint.HTTP2.encode_data/4                                                                                                                                                                                                                                      
            (mint 1.4.2) lib/mint/http2.ex:526: Mint.HTTP2.stream_request_body/3                                                                                                                                                                                                                               
            (k8s 2.0.0) lib/k8s/client/mint/request.ex:196: K8s.Client.Mint.Request.stream_request_body/2                                                                                                                                                                                                      
            (k8s 2.0.0) lib/k8s/client/mint/http_adapter.ex:171: K8s.Client.Mint.HTTPAdapter.handle_call/3                                                                                                                                                                                                     
            (stdlib 4.2) gen_server.erl:1149: :gen_server.try_handle_call/4                                                                                                                                                                                                                                    
            (stdlib 4.2) gen_server.erl:1178: :gen_server.handle_msg/6                                                                                                                                                                                                                                         
            (stdlib 4.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3                                                                                                                                                                                                                                         
    (elixir 1.14.2) lib/gen_server.ex:1038: GenServer.call/3                                                                                                                                                                                                                                                   
    (k8s 2.0.0) lib/k8s/client/mint_http_provider.ex:41: anonymous fn/3 in K8s.Client.MintHTTPProvider.stream/5                                                                                                                                                                                                
    (elixir 1.14.2) lib/stream.ex:1623: Stream.do_resource/5                                                                                                                                                                                                                                                   
    (elixir 1.14.2) lib/stream.ex:942: Stream.do_transform/5                                                                                                                                                                                                                                                   
    (elixir 1.14.2) lib/stream.ex:1811: Enumerable.Stream.do_each/4                                                                                                                                                                                                                                            
    (elixir 1.14.2) lib/stream.ex:942: Stream.do_transform/5                                                                                                                                                                                                                                                   
    (elixir 1.14.2) lib/enum.ex:4307: Enum.each/2                   
```

Reply:
Oh boy yeah that's me being ignorant about iodata / binary / string / charlist.